CMakeLists.txt

```diff
@@ -910,6 +910,13 @@ if(BUILD_TESTS)
     target_include_directories(crypto_test PRIVATE ${CCFCRYPTO_INC})
     target_link_libraries(crypto_test PRIVATE ccfcrypto.host)
 
+    add_unit_test(
+      sharing_test
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/crypto/test/secret_sharing.cpp
+    )
+    target_include_directories(sharing_test PRIVATE ${CCFCRYPTO_INC})
+    target_link_libraries(sharing_test PRIVATE ccfcrypto.host)
```

Comment:
Perhaps re-use the existing `secret_share.cpp` unit test?

Reply:
I think the new one is more exhaustive, and some of the existing checks are no longer applicable because the code no longer splits and instead creates a secret + shares in a single step. There's also no upper limit on the number of shares anymore.

I will add the checks for 0, they do seem potentially valuable (although the governance should prevent that really).